plugins/module_utils/foreman_helper.py

```diff
@@ -1287,6 +1287,40 @@ def entity_name_from_class(self):
         return '_'.join(class_name.split('_')[1:-1])
 
 
+class ForemanInfoAnsibleModule(ForemanStatelessEntityAnsibleModule):
+    """
+    Base class for Foreman info modules that fetch information about entities
+    """
+    def __init__(self, **kwargs):
+        self._resources = []
+        foreman_spec = dict(
+            name=dict(),
+            search=dict(),
+            organization=dict(type='entity'),
+            location=dict(type='entity'),
+        )
+        foreman_spec.update(kwargs.pop('foreman_spec', {}))
+        super(ForemanInfoAnsibleModule, self).__init__(foreman_spec=foreman_spec, **kwargs)
+
+    def run(self, **kwargs):
+        """
+        lookup entities
+        """
+        self.auto_lookup_entities()
+
+        resource = self.foreman_spec['entity']['resource_type']
+
+        if 'name' in self.foreman_params:
```

Comment:
Checking on my pythonic learning, should this be a try-rescue block for a KeyError ?

Reply:
This looks like politely asking if a KeyError would be expected.

Reply:
a try/except block? no-ish.
the code doesn't access `foreman_params['name']`, so there is nothing that would raise the `KeyError`.
(well, `lookup_entity` might at some point, but that's weird)